src/snyk/common/proxy.ts

```diff
@@ -77,3 +93,37 @@ export function getProxyEnvVariable(
   // noinspection HttpUrlsUsage
   return `${protocol}//${auth ? `${auth}@` : ''}${host}${port ? `:${port}` : ''}`;
 }
+
+function getHttpsAgent(configuration: IConfiguration, logger: ILog): Agent {
+  return new Agent({
+    ...getDefaultAgentOptions(configuration, logger),
+  });
+}
+
+function getDefaultAgentOptions(
+  configuration: IConfiguration,
+  logger: ILog,
+  processEnv: NodeJS.ProcessEnv = process.env,
+): AgentOptions | undefined {
+  let defaultOptions: AgentOptions | undefined;
+
+  const sslCheck = !configuration.getInsecure();
+  defaultOptions = { rejectUnauthorized: sslCheck };
+  if (!sslCheck) {
+    globalAgent.options.rejectUnauthorized = false;
```

Comment:
This might mess up other VS Code extensions' `rejectUnauthorized`, since they're sharing the same Node process with our extension. Although I probably wouldn't worry too much, if there's no better way to achieve the same.